src/ALE/regrid_consts.F90

```diff
@@ -8,22 +8,36 @@ module regrid_consts
 
 implicit none ; public
 
-! List of regridding types
-integer, parameter :: REGRIDDING_LAYER     = -1     !< Layer mode
-integer, parameter :: REGRIDDING_ZSTAR     = 0      !< z* coordinates
-integer, parameter :: REGRIDDING_RHO       = 1      !< Target interface densities
-integer, parameter :: REGRIDDING_SIGMA     = 2      !< Sigma coordinates
-integer, parameter :: REGRIDDING_ARBITRARY = 3      !< Arbitrary coordinates
-integer, parameter :: REGRIDDING_HYCOM1    = 4      !< Simple HyCOM coordinates without BBL
-integer, parameter :: REGRIDDING_SLIGHT    = 5      !< Stretched coordinates in the
+integer, parameter :: REGRIDDING_NUM_TYPES  = 7
+
+! List of regridding types. These should be consecutive and starting at 1.
+! This allows them to be used as array indices.
+integer, parameter :: REGRIDDING_LAYER     = 1      !< Layer mode
+integer, parameter :: REGRIDDING_ZSTAR     = 2      !< z* coordinates
+integer, parameter :: REGRIDDING_RHO       = 3      !< Target interface densities
+integer, parameter :: REGRIDDING_SIGMA     = 4      !< Sigma coordinates
+integer, parameter :: REGRIDDING_ARBITRARY = 5      !< Arbitrary coordinates
+integer, parameter :: REGRIDDING_HYCOM1    = 6      !< Simple HyCOM coordinates without BBL
+integer, parameter :: REGRIDDING_SLIGHT    = REGRIDDING_NUM_TYPES !< Stretched coordinates in the
                                                     !! lightest water, isopycnal below
-character(len=5), parameter :: REGRIDDING_LAYER_STRING = "LAYER"   !< Layer string
-character(len=2), parameter :: REGRIDDING_ZSTAR_STRING = "Z*"      !< z* string
-character(len=3), parameter :: REGRIDDING_RHO_STRING   = "RHO"     !< Rho string
-character(len=5), parameter :: REGRIDDING_SIGMA_STRING = "SIGMA"   !< Sigma string
+character(len=6), parameter :: REGRIDDING_LAYER_STRING = "LAYER"   !< Layer string
+character(len=6), parameter :: REGRIDDING_ZSTAR_STRING = "ZSTAR"   !< z* string
+character(len=6), parameter :: REGRIDDING_RHO_STRING   = "RHO"     !< Rho string
```

Comment:
I think I did that because I put these into an array below. My understanding is that Fortran doesn't like having variable length strings in an array. I'll add a comment.